mods/fireflies/init.lua

```diff
@@ -27,6 +27,58 @@ minetest.register_node("fireflies:firefly", {
 	floodable = true,
 	on_flood = function(pos, oldnode, newnode)
 		minetest.add_item(pos, "fireflies:firefly 1")
+	end,
+	on_place = function(itemstack, placer, pointed_thing)
+		local player_name = placer:get_player_name()
+		local pos = pointed_thing.above
+
+		if not minetest.is_protected(pos, player_name) and
+				not minetest.is_protected(pointed_thing.under, player_name) and
+				minetest.get_node(pos).name == "air" then
+			minetest.set_node(pos, {name = "fireflies:firefly"})
+			minetest.get_node_timer(pos):start(1)
+			itemstack:take_item()
+		end
+		return itemstack
+	end,
+	on_timer = function(pos, elapsed)
+		if minetest.get_node_light(pos) > 11 then
+			minetest.set_node(pos, {name = "fireflies:hidden_firefly"})
+		end
+		minetest.get_node_timer(pos):start(30)
+	end
+})
+
+minetest.register_node("fireflies:hidden_firefly", {
+	description = "Hidden Firefly",
+	drawtype = "airlike",
+	inventory_image = "fireflies_firefly.png",
+	wield_image =  "fireflies_firefly.png",
+	paramtype = "light",
+	sunlight_propagates = true,
+	walkable = false,
+	pointable = false,
+	diggable = false,
+	drop = "",
```

Comment:
Probably no needed if not diggable.

Reply:
"drop = ""," is also included by default air, and by every example of an airlike node I can find anywhere.

Reply:
As with 'diggable = false', is harmless and might help in some situations, so it's ok as it is.